### ISSUE TITLE
State of the project

### ISSUE THREAD

**Issue by FagnerMartinsBrack** [impress/impress.js#435]
Hi @bartaz, is this project going to be actively developed?

In several threads you stated no time for working into this project and the latest commits were just regarding grammars.

Impress is impressive (!!), and there are many useful pull requests waiting to be landed. Any news if this project is going somewhere or if you are willing to handle it to someone else to continue active development?

Why do I ask this? Well, maybe I am not the only one, but its sad to see fragmented projects with useful features that could be centralized. If there is any chance that this project is going to get on track, it would be nice to disclose the information, so one may not jump into creating another repository and potentially reinvent the wheel.

Also, I noticed there is a lack of general documentation, so it is hard to know exactly what is supported and what is considered an implementation detail (although the code is pretty well documented and one can have a very good basis).

Anyway, with > 1k watchers and > 22k stars it is unlikely not to exist at least one willing to take over or be an active collaborator. Is there any news regarding the project state? Just to get into the context.

Thanks.

**Comment by FagnerMartinsBrack** [comment]
My current project is not just about a regular slide presentation, it should be scalable and impress.js does not seems to be the best fit for it.

I switched to [jmpress](github.com/jmpress/jmpress.js), a jquery plugin that handles advanced features that impress currently doesnt, with the potential to develop plugins for it. Since I am using angular, I also created an [angular-jmpress prototype](https://github.com/web-stories/angular-jmpress). It is just a prototype of  a basic usage, since I didn't find any repository containing an angular port of jmpress.js or impress.js.

Any help is appreciated. jmpress need contributions and angular-jmpress is just something that, if evolved, could be eventually ported to the jmpress organization.

**Comment by Pierstoval** [comment]
When I'll have more sliders to make, I'll try jmpress thanks for the notice @FagnerMartinsBrack  :)

**Comment by regebro** [comment]
You probably meant to link to https://github.com/jmpressjs/jmpress.js

It's demo page is completely broken on Firefox and Linux, unfortunately, I might look at it when they fix that. :-)

**Comment by FagnerMartinsBrack** [comment]
@regebro Yeah, I fixed the link.

By the way, Firefox is pretty broken for impress itself, at least on Windows. A few days ago Firefox 34.0 wasn't working, several artifacts in the screen when changing a step. It is related to a known issue: https://bugzilla.mozilla.org/show_bug.cgi?id=1098266.

If you are having any troubles at linux you could send a pull request. Like I said, jmpress is a very good initiative with a pretty robust codebase, but need contributions.

**Comment by regebro** [comment]
I should report an issue, you are right.

**Comment by Terrycloth** [comment]
At the very least, it would probably help if the impress.js README was updated with a note about how bartaz can't work on it anymore, and then maybe link here, so users can find forks and plugins if they want to see some of the more recent developments.

**Comment by bobintornado** [comment]
@bartaz maybe the project should adopt [Open-source governance](http://en.wikipedia.org/wiki/Open-source_governance)

**Comment by FagnerMartinsBrack** [comment]
@bobintornado How that type of governance would help this project in its current state? I agree there should be a way to keep the efforts more centralized, if that's what you are suggesting, but I realized the problem here is about maintainers availability and not governance type (this is already MIT licensed so anybody here could take over).

There is also the option for the interested parts to finance this open source project if there's enough ppl willing to pay for it.

**Comment by regebro** [comment]
Open source governance could help by having more maintainers.

**Comment by Terrycloth** [comment]
If we want to do it, then let's just do it. I've started a GitHub Organization -- [impress-js](https://github.com/impress-js) -- so anyone who wants to can participate. For now, anyone who wants to join will be put into a team with write access. If we have enough members, we could hold a small election to see who will get admin access.

**Comment by Pierstoval** [comment]
:+1: I'd like to :)

**Comment by bartaz** [comment]
@Terrycloth there is already an organisation [impress] that I created for the very same reason.
I'm planning to move impress.js repo there and open it up for maintainers.

**Comment by Terrycloth** [comment]
@bartaz 
Oh, I did see the impress username was taken, but it was empty, and there's a user called impressJS who doesn't seem affiliated to impress.js the presentation framework. However you want to do it, then.

**Comment by Pierstoval** [comment]
@bartaz Nice for the @impress organization!
Can't wait! :smile:

**Comment by seyisulu** [comment]
I'd love to join in helping but I the impress and impress-js organizations are empty. How can I assist?

**Comment by FagnerMartinsBrack** [comment]
@seyisulu Unfortunately this thread has been drifting for a while, it's all up to @bartaz to document some API ideas and start triggering the action.

As a sidenote: It would be easier first to document the API with the community participation before changing any code, just a suggestion.

I am also totally up to help contributing here, it is of my interest to have a robust and stable API that could be used to build port for frameworks such as angular and extend for each particular use case. Impress is not just about presentations, it is a framework to create 3D artifacts in an infinite canvas leveraging all HTML5 and CSS3 features.

**Comment by bartaz** [comment]
@seyisulu @FagnerMartinsBrack 

Hi guys, thanks for the interest.
Yes, the impress org is still empty - I want to test the move with some dummy repo before I move whole impress there, but I have other priorities now.

Very good suggestion about documenting API - I guess both in a current state and the 'dream' API with possible use cases.  If any of you have any ideas for that feel free to start writing things down somewhere on the wiki https://github.com/bartaz/impress.js/wiki
Don't worry about the structure too much at the beginning, we can treat it as kind of a brainstorm now and clean it up when we see where it is going.

Thanks!

**Comment by Pierstoval** [comment]
If you just "transfer ownership" to the impress organization, the old repo will redirect to the organization automatically, no worry about the old one as long as you don't recreate a similar repo in your own account

**Comment by bartaz** [comment]
@Pierstoval thanks for confirmation.

Actually I was thinking about forking it back to my account from organisation, to keep 'my copy', but I'm aware that that will break link redirection.

**Comment by Pierstoval** [comment]
Yep it'll break the link, but a simple message in the README.md file might be useful enough to redirect users to the correct repository :)

**Comment by DuncanMacWeb** [comment]
Hey @bartaz, how about transferring ownership of the repo to the impress org *today* and adding @Terrycloth @seyisulu @FagnerMartinsBrack and
@Pierstoval as owners/admins, as they seem motivated to work on it? I think that would be great for the future of impress. [Carpe diem](http://pl.wikipedia.org/wiki/Carpe_diem)! :wink:

**Comment by bartaz** [comment]
Hi guys, things are hopefully going to happen quite soon.

Impress.js will be part of Rails Girls Summer of Code this year, so (assuming that at least one team will be accepted) some work is going to happen during summer time. For sure I hope to get some bugfixes and maybe some new features implemented.

Anyway, for me it will be at least good motivation to move the project to organisation and do some work around it.

So sorry for all the delays and thanks for the patience ;)

**Comment by Pierstoval** [comment]
We'll wait, it's great that it's part of this event!! :+1: :+1:

**Comment by DuncanMacWeb** [comment]
Hey, I think it’s great that impress will be part of RGSoC. Wouldn’t it be a good idea though to move impress into the impress org and invite contributors to be owners of the org now, which would help move the project forward before April turns into May? It would be fantastic to get some significant improvements merged in before RGSoC starts.

@bartaz, about your problem with link redirection if you transfer the repo to the impress org and then fork it back into your account, how about this solution?

1. Transfer the repo to the impress org and add your prospective key contributors as owners
2. Use a different account to fork the repo, or ask someone else to do it (I’d be happy to help with that)
3. Rename the fork while it’s sitting in the other account (`new-impress`, for example)
4. Transfer the fork back into your own account

That way, when the fork is transferred back into your account it would have a different name, so shouldn’t break GitHub’s redirections. I agree that the redirections are very important to maintain existing links etc.

It would be very encouraging to see some progress on impress before RGSoC..... :pray: :smile:

**Comment by bartaz** [comment]
@DuncanMacWeb Just to clarify some things.

Project plan created by RGSoC teams for the applications was based on current state of the project. So (unfortunately) I don't want to many changes to happen before the summer starts. I'm not sure if I'll be able to mentor RGSoC teams when at the same time other changes will be happening in the project.

For me RGSoC is a potential way to move project forward and find future contributors.

As far as I remember in the beginning of May RGSoC teams will be announced, so I'll not if and how many people will be working during the summer and I'll be able to make further plans.

**Comment by henrikingo** [comment]
Some time ago I wanted to up my game when doing talks, and have found impress.js and now I'm using it. Since I use it and I know JavaScript, I'm willing to help. In fact I'm sitting on some to-be-published code already.

*About me:* I'm a good programmer, with knowledge of many languages, including JS. However I'm not so deeply involved in the modern JS toolchains of bower, grunt, cucumber, npm, jquery or jscs, and most importantly I'm totally ignorant of modern JS plugin and dependency injection models. So clearly I cannot help there. (In other words: Javascript frameworks evolve with a higher frequency than I use JavaScript...)

As all of us, I'm busy with family and real work, so my contributions are likely to be spotty, but experience from maintaining a Drupal module for 7 years tells me it still works when we all help each other.

*What I'd like to contribute:*
 - the comments in the demo html source are great, but there really should be real documentation.
 - more examples and templates, specifically, simpler ones than the powerful official demo
 - integrating impress-server and impressConsole.js (specifically, you can enter the impress-server pin via impressConsole.js)

More ideas, to be implemented:
 - integrating the help screen from hovercraft.js (I don't like RST, but some js/css snippets in hovercraft have stayed in my presentations)
 - allow stating step positions as relative to previous step (again, an idea worth keeping from hovercraft)
 - given time, I'd love to integrate a diagram markup language, like mermaid or graphviz

TL;DR: count me in either as maintainer or contributor

**Comment by Klerith** [comment]
Im working in a impress maker, so far so good right now. When is finish I will post it in here, of course is going to be free. 

Right now I can add slides dynamically and move objects. So I thing two or 3 months of hard work and it will be ready to use.

**Comment by Pierstoval** [comment]
@Klerith => Is it already open-sourced? We might all take a look, and you can propose to add this future app in the Impress wiki

**Comment by Klerith** [comment]
As I said before. Is under development. I want to create a website to give every body access to make impress presentations easily and free. 

The code I'm not sure if will be open source, but we will see how it goes. Wish me luck.

**Comment by Pierstoval** [comment]
Good luck then ;)

2015-05-20 15:47 GMT+02:00 Fernando <notifications@github.com>:

**Comment by bartaz** [comment]
Hi folks,
impress.js has now a new home at https://github.com/impress/impress.js 

Also Rails Girls Summer of Code is starting soon, so I hope to make some moves and updates during next couple of months and hopefully this will allow me to make impress more open to contributions and invite some collaborators to the organisation.

**Comment by Pierstoval** [comment]
Yeah, can't wait! :+1:

**Comment by FagnerMartinsBrack** [comment]
:dancer:

**Comment by FagnerMartinsBrack** [comment]
I just want to point out that @bartaz should make it clear what "presentation" means in impress context. I can interpret "presentation" as a bunch of slides that form a website with form controls, etc. Not just public presentation in slides format.

One thing I realized looking through several issues is that the value of impress is not just for static presentations, but to be used as a core framework that uses CSS 3D transform to provide a simple API with the concept of "steps" instead of "slides". A step may contain a slide, or may contain nothing.

The design of the  **public** API must be really low-level, in a way that it could also be conveniently integrated with frameworks like AngularJS.

**Comment by FagnerMartinsBrack** [comment]
Here are some suggestions:

1. Create a principle that express in one or two sentences the purpose of this project, that might close a lot of issues that seems unrelated for what this project specifically is meant to. *"It's a presentation framework based on the power of CSS3 transforms and transitions in modern browsers and inspired by the idea behind prezi.com"* is too broad, that should be the definition of the organization, not the core.
2. Make it clear beforehand that it should be low-level enough for anyone to integrate with angular and other MV* Frameworks or libs (I am familiar with Angular only, sorry) or use it standalone.
3. Create an abstract UML-based architecture to expose the design and let other people suggest enhancements based on specific purposes inside the principle (1) and with practical obvious usage.
4. Create the API docs first to have an idea of how the thing is going to work.
5. In the end, start developing.

Just remember the cost of change, it also applies to OSS projects:

![](http://agilemodeling.com/images/costOfChangeTraditional.gif)

[Here](https://github.com/js-cookie/js-cookie) is an example of what I am talking about, it was continued [from another project](https://github.com/carhartl/jquery-cookie/issues/349), and it's API was [practically rewritten](https://github.com/js-cookie/js-cookie/pull/5) leaving the core intact. The same could be done for impress.js, but with a stronger participation from the community.

**Comment by and3rson** [comment]
@bartaz This is really a very interesting discussion - the moment when the project has to somehow continue its existence and decide a path for future.

Personall I agree @FagnerMartinsBrack that we can make some minor changes to `impress.js` that won't affect its functionality, but rather make it expose its `API` functions to make writing plugins easier.

I would really love this. Having a project with plug-ins is always cool and flexible.

**Comment by bcipolli** [comment]
I just looked over the first page of pull requests, and it seems so easy: a bunch of them are very simple and useful (e.g. `ESC=overview`); others are either nonsense (nonsense changes to `index.html`), complex changes, or not clearly helpful (refactoring the CSS to make it less reusable? Just code churn). Things have to be clearly helpful to take the risk of merging, since there are no regression tests here.

I suggest a few of us simply give a thumbs-up/down + a short 1-2 sentence comment for each PR in the first page. Then @baratz can chime in to either agree or not.

From there, @baratz can either do our bidding (merge or close the PRs), or simply hand the project over to someone who will get community feedback and execute the wishes of the trusted community.

It seems so friggin' easy. Anybody else up for this?

**Comment by and3rson** [comment]
@bcipolli I think multiple PRs may conflict, so the second approach (to hand it to the community) seems more logical. But who will be responsible for it? :)

**Comment by bcipolli** [comment]
Consensus. Yes, @bartaz can follow the consensus, grant access to one person he trusts to follow consensus (goodness, I can do that), or grant access to a few. There aren't all that many people who have shown willingness to participate, and goodness knows it's a thankless job.

From what I see, most of the PRs are noise. Without tests and better coding, most of that stuff has to stay in private repos. But there are simple changes that can definitely be accepted. At least to start and build some trust.

**Comment by and3rson** [comment]
Any updates guys?

It's been a while. A lot of people willing to contribute.

**Comment by bcipolli** [comment]
Again, I think our best way to go is for us to coordinate with comments on specific PRs--thumbs up/down, and a brief sentence why. 

If five of us comment on a single PR, we may be able to convince @baratz to act.

And if we can do it once, perhaps we can build trust to do it twice. Or three times.

I suggest starting with https://github.com/impress/impress.js/pull/399 , which seems simple, uncontroversial, and works great.

**Comment by u982744** [comment]
I think #399 is a bad pull request that should not be merged. Don't get off to a bad start!

**Comment by bcipolli** [comment]
Great to see many people chiming in. I clearly got it wrong, but that worked exactly how it should--group decision and discussion makes things clear. :+1:.

Anybody want to suggest another PR for the group to take a look at?

**Comment by m42e** [comment]
Maybe self advertisement, but I find it useful for a lot of plugins: #235

**Comment by FagnerMartinsBrack** [comment]
Comments were made, no advancements were done.

**Comment by FagnerMartinsBrack** [comment]
@bartaz 

Comments were made, no advancements were done. :cry:

**Comment by and3rson** [comment]
It seems like @bartaz does not really care about his project. It's really weird to behave like that having the repo with 26k and thousands of users.

**Comment by bartaz** [comment]
@FagnerMartinsBrack thanks you for your persistence despite my total lack of response or action.

I'm sorry it turned out this way. Having best intensions was not enough and I'm pretty aware that I failed as a maintainer of this project. It's not true that I don't care about it, but it hasn't been actively developed in almost 4 years and I do have other priorities now.

But it's clear that right now it's me who is blocking any progress here. And the only thing I can do right now to change is to ask others for help.

@FagnerMartinsBrack, @Pierstoval, @bcipolli, @henrikingo - I've noticed you as the most active participants of the discussions around changes in impress.js. I'd like to invite you as contributors to this repository with full rights allowing you to take actions without waiting for me. Before sending the invites I ask only for your confirmation that you will accept it and make a better use of it than me ;)

I hope this step will make it easier for future discussions and actions to take place.

**Comment by Pierstoval** [comment]
@bartaz this is such a honor :+1: 
I confirm I agree with the decision and I think that decisions may be taken with full agreement of the deciders and your policy and goals :)
 
Some other guidelines about the project itself might be good if you have some advices to give, it could help in resolving the tons of awaiting PRs

**Comment by FagnerMartinsBrack** [comment]
:+1: I am definitely willing to contribute and help this project to go forward, since I use it in one of my projects.

**Comment by bartaz** [comment]
@Pierstoval @FagnerMartinsBrack thanks for quick response guys, I'm right now a little lost in GitHub organisation teams/members management so I may have messed something with access rights, but you should get your invitations and have write access to repository.

**Comment by FagnerMartinsBrack** [comment]
Also, we may close this issue once the team is properly setup and once we have made the first relevant change.

My suggestion for the beginning is setting up the codebase for things like:

* Setup a CONTRIBUTING.md file for contribution
* Triaging issues/PRs that are not going to land
* Creating milestones for issues/PRs (keep 0.x.x semver style until we are safe for a 1.0)
* Implement a build system (Grunt? Gulp?)
* Implement linting
* Define a codestyle and enforce it through JSCS
* Create tests
* Setup Travis and maybe Sauce Labs/Karma to run the tests in several browsers

That would put the project in a state of quality and stability that will allow faster and better contribution from other folks.

**Comment by bartaz** [comment]
Great suggestions. I once set up vary basic drone.io build for impress.js just to play with it and get it validated by jshint: https://drone.io/github.com/bartaz/impress.js/admin

**Comment by Pierstoval** [comment]
@FagnerMartinsBrack If you can, you may create different issues for these milestones, but actually, setting up a proper API should be the first thing to do IMO because Impress.js may need a bit of refactoring for it to be more API-compliant. Then, create docs about the API, and after that do everything else :)

**Comment by FagnerMartinsBrack** [comment]
Impress definitely needs an API improvement. The problem is that there are lots of projects/sites that rely on the current code which has no specific API documentation, and therefore testing first will ensure that the behavior will not be drastically changed in an eventual rewrite, a rewrite that could be done with good planning and conservative atomic and revisable changes.

Documentation is another point worth mentioning in the bullet points above, we should be cautious on what to document because documented features will have to be supported until the next major bump.

I just created a [gitter room](https://gitter.im/impress), that is also public, so we can talk more freely about this stuff.

**Comment by bcipolli** [comment]
@FagnerMartinsBrack sure, I would be happy to contribute! Thanks for opening up to move things along. And for sure, setting up tests and making sure that changes are back-compatible is crucial to moving things forward.

**Comment by henrikingo** [comment]
Thank you @bartaz, I'm happy to help. I'm almost as busy as you, but sharing responsibilities is the point of open source.

I also promise to do my best to continue advocating for your vision of impress.js being a tight core library, with more features as plugins, testing, and so on.

While I've been busier than anticipated as well, I've actually been able to take the first little step. Today seems a good day to share what I've been sitting on, so I have made a pull request for adding some first set of qunit testing! (#474) If the pull request is accepted, I will also advocate for a rule that any actual code changes can only be done once the feature in question has code coverage.

Looking at the queue of issues/pull requests, I'd say the first thing to resolve is to add a license file. I will comment on an appropriate issue on that.

**Comment by FagnerMartinsBrack** [comment]
Thanks @henrikingo, I just reviewed your PR and added some interesting suggestions to be discussed. I hope we can move this forward using some help from everyone. Long live open-source!

**Comment by bartaz** [comment]
Couple suggestions from me regarding next steps.

Before introducing any bigger or smaller changes our focus should be on cleaning up current state.
This means (including some of your suggestions):

* Cleaning up Issues and PRs - there over a hundred of them. Sad but true - most of them will probably not go in. Solving bugs and (especially) adding new features should wait until refactoring.
* Adding CONTRIBUTING.md (but I don't have any suggestions what should it say - maybe we can find some example from other repo) and LICENSE.md (with MIT only)
* Cleaning up README.md - at this point it's probably unnecessarily long, maybe it should be split into smaller files (if we want to keep CHANGELOG for example)
* Testing - I always find it harders to 'unit test' such purely visual code like impress.js (verifying that things 'looks' as it should) to it may be tricky, but thanks a lot for the effort already put into it @henrikingo 

And some ideas for the project future:

* independent from other libs - from very beginning impress.js was create without any dependencies on other JS libraries (for DOM or anything else) and it would be great if we could keep that, especially with better and better support for new DOM APIs in newest browsers, which leads us to next point:
* keep browser compatibility to bare minimum - I would put simple and clean code above wide browser compatibility, so my assumption was always to support the newest versions of the browsers assuming they handle 3D transforms (that should be all major these days)
* mobile devices also got much better these days and should handle impress.js, but it's still very hard to predict what can happen - big and image heavy presentations may simply kill mobile browser and I don't know if there is any way it can be 'detected' in advance - that's why iPhone and Android are 'blacklisted' and get fallback only, but maybe it can be changed to be optional
* I know it can be tempting to put a load of tools into the project (build, transpilers, and what not), but I would love to keep impress.js as simple as possible, the 'old school' way: "here is a JS file, add it to HTML, init impress.js and it works"; I know that at least testing may require some dev tooling, but let's keep it to bare minimum
* clean, commented, unminified code - I know that impress.js is not the most basic piece of code, but it still attracts newcomers to web tech, so it would be great to keep it as much learnable as possible

These aren't strict rules by any means, but some points to have in mind when going forward.

**Comment by FagnerMartinsBrack** [comment]
> I know it can be tempting to put a load of tools into the project (build, transpilers, and what not)

@bartaz It is all about quality. If we want to ensure quality I don't mean like transpilers or anything like that. What I mean is to automate and enforce linting, code style and testing. If we want to ensure quality for new contributions we need at least these three. It's all about moving things forward easily and in a robust way, of course never doing more than strictly necessary.

I agree about everything you pointed out, we can follow your ideas for the future of the project and always ping you once something is going to change that.

**Comment by bcipolli** [comment]
>I know it can be tempting to put a load of tools into the project (build, transpilers, and what not)

@FagnerMartinsBrack @bartaz I understood this to mean the tools needed for the end-user, rather than framework tools for contributors. I agree that we need tools for linting/code style/running tests, and we can certainly do that while keeping a single unminified `impress.js` that itself has no dependencies (e.g.. isn't rewritten in coffeescript, isn't using less/scss for styles, etc)

Hopefully I understood correctly!

**Comment by henrikingo** [comment]
@bartaz: My current thinking for the QUnit tests is that we can test that for example a data-x attribute translates to the correct CSS position value. Whether a browser actually renders the thing as expected is out of scope for QUnit. Still, if we get reports that something doesn't render correctly in e.g. Chrome, then we can still add a unit test that checks "don't do X because it's broken in Chrome (version)".

**Comment by GordonSmith** [comment]
Delighted this lib is going to get some fresh work - FWIW I have a tweaked fork which was an experimentation in using impress as a UI - I have a demo here:  http://rawgit.com/GordonSmith/Visualization/WECARE/apps/WeCare/index.html (none of the data will load unless you are inside my intranet)...

**Comment by Pierstoval** [comment]
@GordonSmith Do you have a pending PR from your fork? If you rebase it, we could discuss about the features/fixes you added and make some improvements :wink:

**Comment by GordonSmith** [comment]
@Pierstoval  I don't and wasn't really sure if it the changes were appropriate for a "slide presentation library" (since its basically changing the default actions for space, tab and click) - I am just mentioning it here so folks can think about it during the initial re-factor planning.   
If it helps you planning I can issue a "PR" as a placeholder for the idea?

**Comment by FagnerMartinsBrack** [comment]
@GordonSmith thanks for the link! :D

@Pierstoval 
I am planning to create a script that crawls all forks from a given Github repository with a considerable amount of forks in order to find branches whose revision is ahead of the original. Doing this we can find duplicated features that provide evidence of a popular necessity that might makes sense to be implemented in the main project after some discussion.

For impress.js specifically, I guess it is better to do that after we have a low amount of issues/PR waiting. As it currently stands there is a lot of requested features that are not trivial to implement but are critical to the evolution of impress.js (mostly the ones tagged as "future" a long time ago), so adding more stuff just for the sake of stacking requests will not be of much help right now.

**Comment by AliMD** [comment]
Wow, Goods news
Thank you @FagnerMartinsBrack very much for reactive the community
I'm always love contributing and helping. how can i any help?

**Comment by AliMD** [comment]
I think this issue should be closed too after 2 years.

**Comment by FagnerMartinsBrack** [comment]
> I think this issue should be closed too after 2 years.

I will probably close it eventually once I reach this point of the issues triage :)

**Comment by gitbreaker222** [comment]
just jumping in to tell you _"its great to see this community effort on this project. Thats the spirit of open source software!"_ 👍

**Comment by henrikingo** [comment]
Project is now actively developed.

I will add documentation about how to add tests. Plugin framework is already extensively documented.
